### PR TITLE
Support for `Notebook` CodeAction Kind

### PR DESCRIPTION
**Pull request by Yoyokrazy**
re: #179213 

Add in proposed api for the new CodeActionKind of `Notebook`

Formally support notebook codeactions on save for ipynb notebook files. Notebook Save Participants first call the providers for the first cell of the notebook, running all notebook specific actions across the entire notebook. Once they resolve, providers are called for all cells in the notebook, performing source level actions on each cell in parallel.

As stated in the proposed api, any CodeAction with the `Notebook` will only be called for the first cell of the notebook, and all CodeActions not with that kind must only apply edits to the single cell that they are passed.

### CODE REVIEW

File: src/vs/editor/contrib/codeAction/common/types.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export class CodeActionKind {
 	public static readonly SourceOrganizeImports = CodeActionKind.Source.append('organizeImports');
 	public static readonly SourceFixAll = CodeActionKind.Source.append('fixAll');
 	public static readonly SurroundWith = CodeActionKind.Refactor.append('surround');
+	public static readonly Notebook = new CodeActionKind('notebook');
```
Comment (rebornix) [inline src/vs/editor/contrib/codeAction/common/types.ts:27]: This kind is only used in notebook, should we define this in our notebook component? It feels wrong that monaco needs to know about notebook.

**Review by rebornix — DISMISSED** [review]
Looks good to me in overall, only concern is we are adding the notebook type into Monaco editor. We might want to keep it in `workbench/contrib/notebook`.

File: src/vscode-dts/vscode.proposed.notebookCodeActions.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/179213
+
+	export class NotebookCodeActionKind {
+		// can only return MULTI CELL workspaceEdits
```
Comment (rzhao271) [inline src/vscode-dts/vscode.proposed.notebookCodeActions.d.ts:11]: Debt: could the comment be a docstring?

File: src/vs/workbench/contrib/notebook/browser/contrib/saveParticipants/saveParticipants.ts [hunk 3]
```diff
@@ -152,6 +154,9 @@ class CodeActionOnSaveParticipant implements IStoredFileWorkingCopySaveParticipa
 			});
 		}
 
+
```
Comment (rzhao271) [inline src/vs/workbench/contrib/notebook/browser/contrib/saveParticipants/saveParticipants.ts:157]: Nit: extra spacing

**Review by rzhao271 — APPROVED** [review]

**Pull request merged**